tasks/grafana.py

```diff
@@ -127,9 +128,171 @@ def create_dashboard_in_prod(folder_name, **context):
+
+def backup_dashboard_with_version(dashboard, backup_dir, host, headers, env_name, stored_versions):
+    try:
+        dashboard_uid = dashboard["uid"]
+        dashboard_title = dashboard["title"]
+        
+        # Get the latest version info
+        latest_version = get_latest_version(dashboard_uid, host, headers)
+        if not latest_version:
+            return None
+            
+        current_version = latest_version.get('version')
+        stored_version = stored_versions.get(dashboard_uid)
+
+        # Always backup if this is a new dashboard (no stored version)
+        # Skip only if we have a stored version and it's greater or equal to current
+        if stored_version is not None and stored_version >= current_version:
+            logging.info(f"Skipping {dashboard_title} - no version change (current: {current_version}, stored: {stored_version})")
+            return None
+        
+        response = requests.get(
+            f"{host}/api/dashboards/uid/{dashboard_uid}",
+            headers=headers,
+            verify=False  
+        )
+        response.raise_for_status()
+        dashboard_data = response.json()
+
+        dashboard_data['meta']['version'] = current_version
+        dashboard_data['meta']['message'] = latest_version.get('message', '')
+
+        clean_title = "".join(c if c.isalnum() else "_" for c in dashboard_title)
+        filename = f"{env_name}_{clean_title}_{dashboard_uid}.json"
+        filepath = os.path.join(backup_dir, filename)
+
+        with open(filepath, 'w') as f:
+            json.dump(dashboard_data, f, indent=2)
+        
+        if stored_version is None:
+            logging.info(f"Backed up new dashboard {dashboard_title} - version {current_version}")
+            message = "Initial backup"
+        else:
+            logging.info(f"Backed up {dashboard_title} - version changed from {stored_version} to {current_version}")
+            message = latest_version.get('message', '')
+        
+        return {
+            'filepath': filepath,
+            'version': current_version,
+            'message': message,
+            'title': dashboard_title,
+            'env': env_name,
+            'is_new': stored_version is None
+        }
+    except requests.exceptions.RequestException as e:
+        raise AirflowException(f"Error backing up dashboard {dashboard_title} from {env_name}: {str(e)}")
+
+def backup_all_dashboards(backup_dir, temp_dir, **context):
+    #this function will only process changed and new dashboards
+    os.makedirs(temp_dir, exist_ok=True)
+    
+    try:
+        stored_versions = get_stored_versions(backup_dir)
+        
+        backed_up_files = []
+        changes = []
+        
+        # Backup TEST dashboards
+        test_dashboards = requests.get(
+            f"{TEST_HOST}/api/search?type=dash-db", 
+            headers=HEADERS_TEST, 
+            verify=False
+        ).json()
+        
+        for dashboard in test_dashboards:
```

Comment:
```suggestion
        test_dashboards = requests.get(
            f"{TEST_HOST}/api/search?type=dash-db", 
            headers=HEADERS_TEST, 
            verify=False
        )
        test_dashboards,raise_for_status()
        
        for dashboard in test_dashboards.json():
```
Else it will get ugly if Grafana is not available